src/main/java/com/google/devtools/build/lib/remote/TreeNodeRepository.java

```diff
@@ -264,6 +264,10 @@ private TreeNode buildParentNode(
       int inputsStart,
       int inputsEnd,
       int segmentIndex) {
+    if (segments.isEmpty()) {
+      Preconditions.checkState(inputs.isEmpty());
+      return interner.intern(new TreeNode(ImmutableList.<TreeNode.ChildEntry>of()));
```

Comment:
Can we instead store "new TreeNode(ImmutableList.<TreeNode.ChildEntry>of())" in a static final field and use that instead of calling interner.intern on a new instance everytime?